astropy/convolution/boundary_extend.pyx

```diff
@@ -102,7 +102,7 @@ def convolve2d_boundary_extend(np.ndarray[DTYPE_t, ndim=2] f,
     cdef int wky = nky // 2
```

Comment:
@embray - Yes, ``nx`` cannot be made unsigned since these are compared with ``ii``. ``nkx`` cannot be unsigned because ``wkx`` cannot be unsigned because of ``ii``.

Well they could be if they are converted to integer in operations and comparisons but that wouldn't help much (I think).